haskell/private/cabal_wrapper.sh.tpl

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+#
+# cabal_wrapper.sh <PKG_NAME> <SETUP_PATH> <PKG_DIR> <PACKAGE_DB_PATH> [SETUP_ARGS...]
+
+set -euo pipefail
+execroot=$(pwd)
+
+# Set these variables if unset.
+: ${LD_LIBRARY_PATH:=}
+: ${LIBRARY_PATH:=}
+export LD_LIBRARY_PATH=$execroot/$LD_LIBRARY_PATH
+export LIBRARY_PATH=$execroot/$LD_LIBRARY_PATH
+
+name=$1
+setup=$execroot/$2
+srcdir=$execroot/$3
+pkgroot=$(realpath $execroot/$4/..) # By definition (see ghc-pkg source code).
+shift 4
+
+distdir=$(mktemp -d)
+libdir=$pkgroot/iface
+dynlibdir=$pkgroot/lib
+package_database=$pkgroot/package.conf.d
+
+%{ghc_pkg} recache --package-db=$package_database
+
+# Cabal really wants the current working directory to be directory
+# where the .cabal file is located. So we have no choice but to chance
+# cd into it, but then we have to rewrite all relative references into
+# absolute ones before doing so (using $execroot).
+cd $srcdir
+export HOME=/var/empty
+$execroot/%{runghc} $setup configure \
+    --verbose=0 \
+    --user \
+    --with-compiler=$execroot/%{ghc} \
+    --with-hc-pkg=$execroot/%{ghc_pkg} \
+    --with-ar=%{ar} \
+    --with-strip=%{strip} \
+    --enable-deterministic \
+    --enable-relocatable \
+    --builddir=$distdir \
+    --prefix=$pkgroot \
+    --libdir=$libdir \
+    --dynlibdir=$dynlibdir \
+    --libsubdir= \
+    --package-db=clear \
+    --package-db=global \
+    "${@/=/=$execroot/}" \
+    --package-db=$package_database # This arg must come last.
```

Comment:
I think all of these should be quoted as well.

Reply:
I normally quote things nonetheless (because you cannot distinguish between bugs and known implicit guarantees and explicit quoting is the way to work around the biggest `sh` misfeature)